crates/provider/src/shared/types.rs

```diff
@@ -0,0 +1,185 @@
+use std::{
+    collections::HashMap, fs::File, io::Write, os::unix::process::ExitStatusExt, path::PathBuf,
+    process::ExitStatus, vec,
+};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum ZombieRole {
+    Temp,
+    Node,
+    BootNode,
+    Collator,
+    CumulusCollator,
+    Authority,
+    FullNode,
```

Comment:
Are all these types used @pepoviola because from what I've seen in the typescript version, there is an overlap between ```label``` and ```role``` in the same enum but not all variant are really used for ```role```